README.md

```diff
@@ -2,107 +2,112 @@
 
 Similarly, multiple roles can be granted, revoked or checked for in a single transaction using bitwise operations:
 - granting role(s) is a bitwise AND operation of the currently assigned value and the value of the new role(s);
 - revoking role(s) is a bitwise AND operation of the currently assigned value and the complement (bitwise NOT) of the value of the role(s) to be revoked; and
 - checking for membership is a bitwise OR operation of the currently assigned value and the value of the role(s) being checked for.
 
-Note that functions which rely on this access control library will require the `bitwise_ptr` implicit argument and `BitwiseBuiltin`.
-
-
 ## Usage
 
-To use this library in a Cairo contract:
-1. Include a copy of `accesscontrol_library.cairo` in your project, and import the library into the Cairo contract.
-2. Define the available roles as constants in a namespace in a separate Cairo contract, and import this namespace into the Cairo contract.
+To use this library, add the repository as a dependency in your `Scarb.toml`:
 
-For example, assuming you have a `contracts/` folder with `accesscontrol_library.cairo` and `roles.cairo`, and you want to import both into a Cairo file within the same folder:
+```
+[dependencies]
+wadray = { git = "https://github.com/lindy-labs/cairo-wadray.git" }
+```
 
+Next, define the available roles in a separate Cairo file:
 ```cairo
-%lang starknet
-
-from starkware.cairo.common.cairo_builtins import BitwiseBuiltin, HashBuiltin
-from contracts.accesscontrol_library import AccessControl
-from contracts.roles import Roles
-
-@view
-func is_manager{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}(user: felt) -> (authorized: felt) {
-    let authorized: felt = AccessControl.has_role(Roles.MANAGER, user);
-    return (authorized,);
+mod user_roles {
+    const MANAGER: u128 = 1;
+    const STAFF: u128 = 2;
+    const USER: u128 = 4;
 }
+```
+then import both the component and the roles into your Cairo contract.
+
+For example, assuming you have a `src/` folder with the roles defined in a `user_roles` module in `roles.cairo`:
+```
+use starknet::ContractAddress;
 
-@external
-func authorize{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(
-    role: felt, user: felt
-) {
-    AccessControl.assert_admin();
-    AccessControl._grant_role(role, user);
-    return ();
+#[starknet::interface]
+trait IMockContract<TContractState> {
+    fn is_manager(self: @TContractState, user: ContractAddress) -> bool;
 }
 
-@external
-func manager_only_action{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*}() {
-    AccessControl.assert_has_role(Roles.MANAGER);
-    // Insert logic here
-    return ();
+#[starknet::contract]
+mod mock_contract {
+    use access_control::access_control::access_control_component;
+    use src::roles::user_roles;
```

Comment:
The `src` wouldn't work here, root is based on the project name in Scarb.toml if I'm not mistaken. We can just use `my_project::roles::user_roles`.